content-src/components/DiscoveryStreamComponents/Hero/Hero.jsx

```diff
@@ -34,7 +34,21 @@ export class Hero extends React.PureComponent {
       );
     }
 
-    let [heroRec, ...otherRecs] = data.recommendations.slice(0, this.props.items);
+    const {spocs} = this.props || {};
+    const recs = data.recommendations.slice(0, this.props.items);
+
+    if (spocs.positions && spocs.positions.length) {
+      for (let position of spocs.positions) {
+        const {result} = position;
+        if (result) {
+          // Insert spoc into the desired index and pop the last item.
+          recs.splice(position.index, 0, result);
+          recs.pop();
+        }
+      }
+    }
```

Comment:
Let's move this block up into `DiscoveryStreamBase.jsx` instead of putting copies of it in both components.